tests/spin/test_lattice.py

```diff
@@ -525,6 +525,231 @@ def test_add_edge():
                 (8, 9, 0),
             ],
         ),
+        (
+            "LIEB",
```

Comment:
Do we just test the edges of the lattice? How about the correctness of other attributes?

Reply:
We can test number of sites, or coordinates for lattices. I kept this suite limited as we already tested that when adding tests for the lattice class itself.

Reply:
So the correctness of the lattice object is tested for all lattice types? If yes, then the current test is enough.

Reply:
Added more tests to cover the templates which were missing.